src/relay/op/nn/convolution.cc

```diff
@@ -248,10 +248,17 @@ bool Conv2DTransposeRel(const Array<Type>& types,
   }
   // dilation
   Array<IndexExpr> oshape({dshape_nchw[0], channels, 0, 0});
-  oshape.Set(2, (param->strides[0] * (dshape_nchw[2] - 1) + dilated_ksize_y -
-                 2 * param->padding[0] + param->output_padding[0]));
-  oshape.Set(3, (param->strides[1] * (dshape_nchw[3] - 1) + dilated_ksize_x -
-                 2 * param->padding[1] + param->output_padding[1]));
+  if ( param->padding.size() == 2 ) {
+    oshape.Set(2, (param->strides[0] * (dshape_nchw[2] - 1) + dilated_ksize_y -
+                   2 * param->padding[0] + param->output_padding[0]));
+    oshape.Set(3, (param->strides[1] * (dshape_nchw[3] - 1) + dilated_ksize_x -
+                   2 * param->padding[1] + param->output_padding[1]));
+  } else if (param->padding.size() == 4) {
+    oshape.Set(2, (param->strides[0] * (dshape_nchw[2] - 1) + dilated_ksize_y -
+                   param->padding[0] - param->padding[2] + param->output_padding[0]));
+    oshape.Set(3, (param->strides[1] * (dshape_nchw[3] - 1) + dilated_ksize_x -
+                   param->padding[1] - param->padding[3] + param->output_padding[1]));
```

Comment:
Formulas are very similar except of  using 
```
2 * param->padding[0]  vs  param->padding[0] + param->padding[2]
and 
2 * param->padding[1]   vs   param->padding[1] + param->padding[3]
```
Can we calculate correct padding and then use the same formulas with calculated padding for both cases?

Reply:
The additional condition is to handle the head padding and tail padding are diffenct, for even kernel in this case. I didn't quite understand your point. Please clearify.

Reply:
What about the following?
```
  if (param->padding.size() == 2) {
    hpad = 2 * param->padding[0];
    wpad = 2 * param->padding[1];
  } else if (param->padding.size() == 4) {
    hpad = param->padding[0] + param->padding[2];
    wpad = param->padding[1] + param->padding[3];
  }
  oshape.Set(2, (param->strides[0] * (dshape_nchw[2] - 1) + dilated_ksize_y -
                 hpad + param->output_padding[0]));
  oshape.Set(3, (param->strides[1] * (dshape_nchw[3] - 1) + dilated_ksize_x -
                 wpad + param->output_padding[1]));
```

Reply:
This is better.  Thanks! 
But the input padding tuple is (top, left, bottom, right).  So, 
```
  } else if (param->padding.size() == 4) {
    pad_h = param->padding[0] + param->padding[2];
    pad_w = param->padding[1] + param->padding[3];
  } else {
```